ncbi/src/main/java/org/reactome/release/EuropePMC.java

```diff
@@ -0,0 +1,375 @@
+package org.reactome.release;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.neo4j.driver.v1.Session;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.reactome.release.Utilities.appendWithNewLine;
+
+/**
+ * File generator for Europe PMC.  This class will produce files for
+ * Europe PMC profile XML describing Reactome as a data provider and
+ * Europe PMC link XML describing Reactome pathways annotated directly
+ * (i.e. not on parent or child pathways) with PubMed literature references.
+ * @author jweiser
+ */
+public class EuropePMC {
+	private static final Logger logger = LogManager.getLogger();
+	private static final String rootTag = "links";
+	private static final int reactomeProviderID = 1903;
+
+	private String outputDir;
+	private int reactomeVersion;
+
+	/**
+	 * Returns a new instance of the Europe PMC File Generator
+	 * @param outputDir Directory path for output files
+	 * @param reactomeVersion Release version of Reactome
+	 */
+	public static EuropePMC getInstance(String outputDir, int reactomeVersion) {
+		return new EuropePMC(outputDir, reactomeVersion);
+	}
+
+	private EuropePMC(String outputDir, int reactomeVersion) {
+		this.outputDir = outputDir;
+		this.reactomeVersion = reactomeVersion;
+	}
+
+	/**
+	 * Queries graph database for all human pathways with literature references
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Set of Europe PMC Link objects describing the pathway to literature reference annotations
+	 */
+	public static Set<EuropePMCLink> fetchEuropePMCLinks(Session graphDBSession) {
+		logger.info("Fetching Europe PMC Links from Reactome Pathways");
+
+		Set<EuropePMCLink> europePMCLinks = graphDBSession.run(
+			String.join(System.lineSeparator(),
+				"MATCH (p:Pathway)-[:literatureReference]->(lr:LiteratureReference)",
+				"WHERE p.speciesName = 'Homo sapiens' AND lr.pubMedIdentifier IS NOT NULL",
+				"RETURN DISTINCT p.displayName, p.stId, lr.pubMedIdentifier",
+				"ORDER BY p.displayName"
+			)
+		)
+		.stream()
+		.map(record ->
+			new EuropePMCLink(
+				record.get("p.displayName").asString(),
+				record.get("p.stId").asString(),
+				Integer.toString(record.get("lr.pubMedIdentifier").asInt())
+			)
+		).collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
What is this `::new` notation?

Reply:
A new thing in Java 8, still not in very common usage (but becoming more common):
https://stackoverflow.com/questions/20001427/double-colon-operator-in-java-8

Reply:
Yeah, it's a method reference to the constructor to get a new object (used instead of a lamba which would use the traditional syntax with the new keyword).  It creates a new LinkedHashSet to be filled with the results of the stream.